test/src/discover_tasks_test.dart

```diff
@@ -147,6 +148,40 @@ main() {
     });
   });
 
+  group('execute returns value', () {
+    grinderTest('task from method', () {}, (MockGrinderContext ctx) {
+      var annotated =
+          discoveryGood.discoverDeclaration(goodLib.declarations[#method], {});
+      var task = annotated.task;
+      expect(task.name, 'method');
```

Comment:
The intention was not to test the discovery but whether the return value was handled properly. I tried to cover the three possible paths.
At first I intended to just add two lines (execute the discovered tasks and evaluate the result) where discovery was tested, but with the need of `GrinderContext` this seemed to add too much noise and disguise the main intent of the tests. 
Is there a way to simplify these added tests? I haven't played with imperatively adding tasks to a `Grinder` instance yet.